azurerm/resource_arm_kubernetes_cluster.go

```diff
@@ -1529,3 +1586,20 @@ func flattenKubernetesClusterKubeConfigAAD(config kubernetes.KubeConfigAAD) []in
 		},
 	}
 }
+
+func flattenKubernetesClusterManagedClusterIdentity(input *containerservice.ManagedClusterIdentity) []interface{} {
+	if input == nil {
+		return []interface{}{}
+	}
+
+	identity := make(map[string]interface{})
+	if input.PrincipalID != nil {
```

Comment:
I think with .12 we need to ensure these are set now?
```suggestion
	input.PrincipalID = ""
	if input.PrincipalID != nil {
```